src/sync/sync_config.hpp

```diff
@@ -43,7 +45,11 @@ using SyncBindSessionHandler = void(const std::string&,          // path on disk
                                     std::shared_ptr<SyncSession> // the session which should be bound.
                                     );
 
-using SyncSessionErrorHandler = void(std::shared_ptr<SyncSession>, int error_code, std::string message, SyncSessionError);
+using SyncSessionErrorHandler = void(std::shared_ptr<SyncSession>,
+                                     std::error_code error_code,
+                                     bool is_fatal,
```

Comment:
I think that makes sense. With the new `sync` API changes, there's less of a need for the enum.